### ISSUE TITLE
When to start an f3 instance

### ISSUE THREAD

**Issue by ranchalp** [filecoin-project/go-f3#173]
Once an f3 instance terminates, we need to determine when the next one starts. This decision will influence the head input chain offset #175 (waiting longer increases likelihood of preagreement on start for a particular head input offset).

**Comment by ranchalp** [comment]
A decision was made to wait for some delay after the timestamp of the last finalized epoch by the terminated f3 instance - parameter to be decided for FIP. 

For the moment, let us assume to wait 2 epochs from latest finalized epoch. That is, a participant, on termination of an f3 instance: 

- looks at the epoch $e$ of the head of the newly finalized prefix. 
- looks at the current epoch $e'$
- if $e'-e$>= 2 then start immediately. Otherwise, wait until epoch e+2 starts to start a new f3 instance.

This is in pair with the decision for #175 to suggest as proposal for the new instance the current EC heaviest chain from baseChain up until the second head (i.e. removing the head tipset from the proposal), but suggesting at least the tipset from epoch e+1.

**Comment by Stebalien** [comment]
TL;DR: will tell go-f3 when to start which may be "now" or at some future time T when we expect the next block.

**Comment by Stebalien** [comment]
However, if we see an "empty" instance, we need to make sure we don't start racing. Either:

1. Add some kind of exponential backoff.
2. Wait until `LastTipsetTime + BlockTime * EmptyInstanceCount`.

**Comment by jennijuju** [comment]
(work here will happen in the client, which will feed it into the f3

**Comment by anorth** [comment]
Closing this as done, carrying over the question about racing to a new issue https://github.com/filecoin-project/go-f3/issues/371
